task1.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+let base = 1000000;
+
+function addition(arr1, arr2) {
+  if (arr1.length < arr2.length) {
+    [arr1, arr2] = [arr2, arr1];
+  }
+  if (arr2[0] === 0 && arr2.length === 1) return arr1;
+  let len1 = arr1.length;
+  let len2 = arr2.length;
+
+  for (let i = 0; i < len2; i++) {
+    arr1[i] += arr2[i];
+    if (arr1[i] >= base) {
+      arr1[i] -= base;
+      i === len1 - 1 ? arr1.push(1) : (arr1[i + 1] += 1);
+    }
+  }
+  return arr1;
+}
+
+function multiplication(arr1, arr2) {
+  if (!arr1[0] && arr1.length === 1) return arr1;
+  let len1 = arr1.length;
+  let len2 = arr2.length;
+
+  if (len1 !== len2) {
+    if (len1 < len2) {
+      while (arr2.length !== len1) {
+        arr1.push(0);
+      }
+    } else {
+      while (arr2.length !== len1) {
+        arr2.push(0);
+      }
+    }
+  }
+
+  let res = new Array(len2 + len1 - 1).fill(0);
+  let resLen = res.length - 1;
+  for (let i = 0; i < len1; i++) {
```

Comment:
```suggestion
  for (let i = 0; i < len2; i++) {
```

Reply:
Are you going to apply this suggestion or is it unnecessary?

Reply:
Concreate for Fibonacci task `len1` and` len2` are equal. So I stashed them and declare an additional variable `len`

Reply:
True. But I think that the long multiplication should be more abstract and work with arrays of different sizes.